src/shared/messaging/test/port-rpc-test.js

```diff
@@ -175,23 +175,158 @@ describe('PortRPC', () => {
     assert.calledWith(testMethod, 'second', 'call');
   });
 
-  describe('#disconnect', () => {
-    it('disconnects and returns port', async () => {
-      const rpc = new PortRPC();
-      const testMethod = sinon.stub();
-      rpc.on('test', testMethod);
+  it('should send "connect" event after connection', async () => {
+    const { port1, port2 } = new MessageChannel();
+    const sender = new PortRPC();
+    const receiver = new PortRPC();
+    const connectHandler = sinon.stub();
+
+    receiver.on('connect', connectHandler);
+    receiver.connect(port2);
+    sender.connect(port1);
+
+    await waitForMessageDelivery();
+
+    assert.calledWith(connectHandler);
+  });
+
+  it('should send "close" event after port is destroyed', async () => {
+    const { port1, port2 } = new MessageChannel();
+    const sender = new PortRPC();
+    const receiver = new PortRPC();
+    const closeHandler = sinon.stub();
+
+    receiver.on('close', closeHandler);
+    receiver.connect(port2);
+    sender.connect(port1);
+    await waitForMessageDelivery();
+
+    closeHandler.resetHistory();
```

Comment:
I believe this line should not be here. `closeHandler` handler should be invoked only once.
```suggestion
```

Reply:
The purpose of this was to make sure that `close` is definitely dispatched in response to `sender.destroy()` and not any earlier. I changed this to an assert to be clearer about that. Same for other `closeHandler.resetHistory` calls.